dace/transformation/dataflow/redundant_array.py

```diff
@@ -154,6 +156,63 @@ def can_be_applied(graph, candidate, expr_index, sdfg, strict=False):
             if any(m != a
                    for m, a in zip(edge.data.subset.size(), in_desc.shape)):
                 return False
+            # In strict mode, check if the state has two or more access nodes
+            # for the output array. Definitely one of them (out_array) is a
+            # write access. Therefore, there might be a RW, WR, or WW dependency.
+            accesses = [n for n in graph.nodes()
+                        if isinstance(n, nodes.AccessNode) and
+                        n.desc(sdfg) == out_desc and n is not out_array]
+            if len(accesses) > 0:
+                # We need to ensure that a data race will not happen if we
+                # remove in_array.
+                # First, we simplify the graph (see transient_reuse)
+                # Copy the whole graph
+                G = nx.MultiDiGraph()
```

Comment:
could we refactor this (graph creation part) to a separate function in dace.transformation.helpers or dace.sdfg.utils?